### PR TITLE
cmake: Install json/python files

### LINKED ISSUE #372: Add installation code
On Windows there are currently no install targets.

Here are all the files that need to be installed:

- [x] Bin\VkLayer_khronos_profiles.dll
- [x] Bin\VkLayer_khronos_profiles.json
- [x] Config\VK_LAYER_KHRONOS_profiles\VP_ANDROID_baseline_2021.json
- [x] Config\VK_LAYER_KHRONOS_profiles\VP_ANDROID_baseline_2022.json
- [x] Config\VK_LAYER_KHRONOS_profiles\VP_KHR_roadmap_2022.json
- [x] Config\VK_LAYER_KHRONOS_profiles\VP_LUNARG_desktop_baseline_2022.json
- [x] Config\VK_LAYER_KHRONOS_profiles\VP_LUNARG_desktop_baseline_2023.json
- [x] Config\VK_LAYER_KHRONOS_profiles\VP_LUNARG_desktop_portability_2022.json
- [x] share\vulkan\registry\gen_profiles_file.py
- [x] share\vulkan\registry\gen_profiles_solution.py
- [x] share\vulkan\registry\profiles-0.8-latest.json
- [x] include/vulkan/vulkan_profiles.hpp

vulkan_profiles.h and vulkan_profiles.hpp should NOT be included!

### PR DESCRIPTION
**Pull request by juan-lunarg**
The relevant json and python files are now being installed to the proper locations to match the SDK build process.

closes #372

### CODE REVIEW

**Comment by juan-lunarg** [comment]
This is what installation of profiles looks like now:

```
juan@pop-os:~/projects/profiles$ cmake --install build/ --prefix build/install
-- Install configuration: "Debug"
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/config/VK_LAYER_KHRONOS_profiles/VP_ANDROID_baseline_2021.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/config/VK_LAYER_KHRONOS_profiles/VP_ANDROID_baseline_2022.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/config/VK_LAYER_KHRONOS_profiles/VP_KHR_roadmap_2022.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/config/VK_LAYER_KHRONOS_profiles/VP_LUNARG_desktop_baseline_2022.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/config/VK_LAYER_KHRONOS_profiles/VP_LUNARG_desktop_baseline_2023.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/config/VK_LAYER_KHRONOS_profiles/VP_LUNARG_desktop_portability_2022.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/registry/gen_profiles_file.py
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/registry/gen_profiles_solution.py
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/registry/profiles-0.8-latest.json
-- Up-to-date: /home/juan/projects/profiles/build/install/share/vulkan/explicit_layer.d/VkLayer_khronos_profiles.json
-- Up-to-date: /home/juan/projects/profiles/build/install/lib/libVkLayer_khronos_profiles.so
```

File: profiles/CMakeLists.txt [hunk 1]
```diff
@@ -146,12 +150,33 @@ add_dependencies(VpTestUnion VpGenerated)
...
+)
+
+install(
+    FILES
+        ${MERGE_PYTHON_FILES}
+        ${LAYER_PYTHON_FILES}
+        ${PROJECT_SOURCE_DIR}/schema/${PROFILES_SCHEMA_FILENAME}
+    DESTINATION
+        "${CMAKE_INSTALL_DATADIR}/vulkan/registry"
+)
```
Comment (juan-lunarg) [inline profiles/CMakeLists.txt:182]: Double check this code in particular.

File: CMakeLists.txt [hunk 2]
```diff
@@ -182,16 +182,11 @@ endif()
 
 set(PROFILES_SCHEMA_FILENAME "profiles-0.8-latest.json")
 
-# The scripts directory regenerates the Profiles source and headers.
-add_subdirectory(scripts)
+# The profiles directory regenerates the Profiles source and headers.
+add_subdirectory(profiles)
```
Comment (juan-lunarg) [inline CMakeLists.txt:186]: Moving the logic to profiles made the installation code easier to write. And IMO reads better.

**Review by christophe-lunarg — CHANGES_REQUESTED** [review]
`vulkan_profiles.hpp` should be included, not `vulkan_profiles.h`

File: library/CMakeLists.txt [hunk 3]
```diff
@@ -14,3 +14,11 @@ endif()
...
 target_include_directories(VulkanProfiles INTERFACE include)
 add_library(Vulkan::Profiles ALIAS VulkanProfiles)
+
+# NOTE: vulkan_profiles.h should NOT be installed!
+install(
+    FILES
+        ${CMAKE_SOURCE_DIR}/library/include/vulkan/debug/vulkan_profiles.hpp
+    DESTINATION
+        ${CMAKE_INSTALL_INCLUDEDIR}/vulkan/
+)
```
Comment (juan-lunarg) [inline library/CMakeLists.txt:24]: Updated based on latest feedback

**Review by christophe-lunarg — APPROVED** [review]

**Pull request merged**